### DISCUSSION TITLE
Add Comments to Code and Layout Files

### DISCUSSION THREAD

**Discussion by isiko** [EventFahrplan/EventFahrplan#452] [Q&A]
It would be very helpful if the code was better commented to help find what you are looking for. A possible conventions would be:

- One comment at the beginning of each file describing the purpose of the content.
- One comment per method and attribute explaining the purpose of the content.

**Comment by johnjohndoe** [comment]
Thank you for pointing this out. It can be difficult to notice this once you worked on a project for a while. :see_no_evil: 

Can you please point out concrete examples of where you got stuck? I ask because the overall attempt is to use package, class, function, field and variables names which are **self-descriptive**. If that is not the case it is better to **change** the name instead of adding a comment. There are of course portions of code which need comments because the **intention** of the author, an **algorithm** needs explanation or one wants to document something which is **not written** in code. I would however not put comments in general but **in specific places** if needed. This is also because comments tend to date out very quickly.

Please do not take my reply as a general "no" to your request. I appreciate your idea and I want to make it comfortable to get started on the project for new people. :+1:

**Comment by johnjohndoe** [comment]
:bulb: If the initial overview in the project is the challenge then I am happy to give you a **guided tour** :footprints: through code base of the project at some point via screen sharing. Let me know.

**Comment by isiko** [comment]
First of all, thanks for the quick reply! I also think that self-describing code is often much better than adding a whole novel to each method.

So far I've only looked at the layout side of the project, so I can't say anything about the code, but one example would be the multiple layout files for the details fragment.It's not immediately clear why 5 layout files that have almost the same content are needed, I just found out when I tested my changes for #441 on a tablet emulator where the 'narrow' version was used.

There are probably more examples like this, but I haven't really gotten involved in this project until now (and probably part of my confusion comes from having only half a year of Android development experience), I just had a lot of trouble at the beginning and thought about what might make it easier for future developers. 

((Apart from that, it also doesn't help that Android Studio just puts all layout files into a list without any possibility for packages or folders, but that's a problem with Android Studio, not with this project ☹️))

**Comment by johnjohndoe** [comment]
I am happy to clarify your findings as far as I can:

- There are different layout files for smartphones and tablets. That is perfectly correct. If you look for their usage then you will find the code location where it is decided which one is used.
- There are even more layout files in `layout`, `layout_port`,  `values-sw600dp`, `values-xlarge`, ... which is **truly confusing** if you are familiar with the [standard layout system](https://developer.android.com/guide/topics/large-screens/support-different-screen-sizes) which Android provides. This is **a relikt** of the former developer probably not being informed about this mechanism. The current implementation kind of circumvents the layout system. I do not know the intention behind this implementation. It is just there because nobody took the time to **clean it up**. This should definitely happen.
- Android Studio uses the `Android` perspective in the `Project` panel since a few years as the default for new projects. Please try the `Project` perspective which shows the natural structure of the folders tree.
  ![project](https://user-images.githubusercontent.com/144518/150021947-66667ec9-712e-4f99-9390-fd986fe80686.png)

_You got a few duplications in your reply above. You might want to edit it to clean it up._
